sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeTest.java

```diff
@@ -671,6 +671,22 @@ public void showMandatoryDownloadReadyDialogTest() {
         verify(mReleaseDownloader, times(2)).resume();
     }
 
+    @Test
+    public void tryResetWorkflowWhenApplicationEnterForegroundWhenChannelNull() {
+        Distribute.getInstance().onStarting(mAppCenterHandler);
+        Distribute.getInstance().onApplicationEnterForeground();
+        verifyStatic(never());
+        DistributeUtils.getStoredDownloadState();
+    }
+
+    @Test
+    public void tryResetWorkflowWhenApplicationEnterForegroundWhenChannelNotNull() {
+        start();
+        Distribute.getInstance().onApplicationEnterForeground();
+        verifyStatic();
+        DistributeUtils.getStoredDownloadState();
+    }
```

Comment:
I don't see any meaningful check that it will be check updates on enter foreground.

Reply:
The test should test that the first automatic call is made, then background, then foreground, then another call is made.